src/main/java/life/qbic/openbis/openbisclient/OpenBisClient.java

```
@@ -1023,14 +1031,20 @@ public Vocabulary getVocabulary(String vocabularyCode) {
    */
   @Override
   public List<Experiment> getExperimentsForUser(String userID) {
-    ensureLoggedIn();
     loginAsUser(userID);
-    SearchResult<Experiment> experiments = v3.searchExperiments(sessionToken,
-        new ExperimentSearchCriteria(), fetchExperimentsCompletely());
-
+    // we are not reusing other functions to be sure the user in question is actually used
+    SearchResult<Experiment> experiments = null;
+    try {
+      experiments = v3.searchExperiments(sessionToken, new ExperimentSearchCriteria(),
+          fetchExperimentsCompletely());
+    } catch (UserFailureException u) {
+      logger.error("Could not fetch experiments for user " + userID
+          + ", because they could not be logged in. Is user " + this.userId + " an admin user?");
+      logger.warn("No experiments were returned.");
+    }
     logout();
     login();
-    if (experiments.getObjects().isEmpty()) {
+    if (experiments == null || experiments.getObjects().isEmpty()) {
       return null;
```

Maybe throw an exception here as well?